extension/wasm/wasm_bindings.cpp

```diff
@@ -0,0 +1,561 @@
+
+  const Tensor& get_tensor() const {
+    THROW_IF_FALSE(tensor_, "Tensor is null");
+    return *tensor_;
+  }
+
+  ScalarType get_scalar_type() const {
+    THROW_IF_FALSE(tensor_, "Tensor is null");
+    return tensor_->scalar_type();
+  }
+  val get_data() const {
+    switch (get_scalar_type()) {
+#define JS_CASE_TENSOR_TO_VAL_TYPE(T, NAME)                        \
+  case ScalarType::NAME:                                           \
+    THROW_IF_FALSE(tensor_->data_ptr<T>(), "Tensor data is null"); \
+    return val(typed_memory_view(tensor_->numel(), tensor_->data_ptr<T>()));
+      JS_FORALL_SUPPORTED_TENSOR_TYPES(JS_CASE_TENSOR_TO_VAL_TYPE)
+      default:
+        THROW_JS_ERROR(
+            TypeError, "Unsupported Tensor type: %d", get_scalar_type());
+    }
+  }
+  val_array<int> get_sizes() const {
+    return val::array(get_tensor().sizes().begin(), get_tensor().sizes().end());
+  }
+
+  static std::unique_ptr<JsTensor>
+  full(val_array<int> sizes, val fill_value, val type = val::undefined()) {
+    auto sizes_vec =
+        convertJSArrayToNumberVector<executorch::aten::SizesType>(sizes);
+    ScalarType scalar_type =
+        type.isUndefined() ? ScalarType::Float : type.as<ScalarType>();
+    switch (scalar_type) {
+#define JS_CASE_FULL_VECTOR_TYPE(T, NAME)                                 \
+  case ScalarType::NAME: {                                                \
+    TensorPtr tensor =                                                    \
+        extension::full(sizes_vec, fill_value.as<T>(), ScalarType::NAME); \
+    return std::make_unique<JsTensor>(std::move(tensor));                 \
+  }
+      JS_FORALL_SUPPORTED_TENSOR_TYPES(JS_CASE_FULL_VECTOR_TYPE)
+      default:
+        THROW_JS_ERROR(TypeError, "Unsupported Tensor type: %d", scalar_type);
+    }
+  }
+
+  static std::unique_ptr<JsTensor> zeros(
+      val_array<int> sizes,
+      val type = val::undefined()) {
+    auto sizes_vec =
+        convertJSArrayToNumberVector<executorch::aten::SizesType>(sizes);
+    ScalarType scalar_type =
+        type.isUndefined() ? ScalarType::Float : type.as<ScalarType>();
+    TensorPtr tensor = extension::zeros(sizes_vec, scalar_type);
+    return std::make_unique<JsTensor>(std::move(tensor));
+  }
+
+  static std::unique_ptr<JsTensor> ones(
+      val_array<int> sizes,
+      val type = val::undefined()) {
+    auto sizes_vec =
+        convertJSArrayToNumberVector<executorch::aten::SizesType>(sizes);
+    ScalarType scalar_type =
+        type.isUndefined() ? ScalarType::Float : type.as<ScalarType>();
+    TensorPtr tensor = extension::ones(sizes_vec, scalar_type);
+    return std::make_unique<JsTensor>(std::move(tensor));
+  }
+
+  static std::unique_ptr<JsTensor> from_array(
+      val_array<int> sizes,
+      val_array<val> data,
+      val type = val::undefined(),
+      val_array<int> dim_order = val::undefined(),
+      val_array<int> strides = val::undefined()) {
```

Comment:
Do some checks such as dim_order and strides have appropriate sizes, and dim_order contains unique values in valid rang